external-crates/move/crates/move-package/src/compilation/compiled_package.rs

```diff
@@ -415,6 +416,7 @@ impl CompiledPackage {
             /* source paths */ Vec<Symbol>,
             /* address mapping */ &ResolvedTable,
             /* compiler config */ PackageConfig,
+            /* whether source is available */ bool,
```

Comment:
Acknowledging this commit is a cherry-pick of existing code, it might be an improvement to represent this as an `enum DependencyKind`  representing `Bytecode` versus `Source` dependencies. It could make it more clear here, and in control flow, e.g., in the `partition_map` logic, for example, what `b` represents: 

```
.partition_map(|(p, b)| if b { Either::Left(p) } else { Either::Right(p) });
```

Reply:
I wanted to keep it the same as upstream because at the time I created the commit I was considering the upstreamability of the changes. But seeing that the codebases have anyways diverged and are not being kept in sync I'm ok to change this per your suggestion.